pkg/backends/calico/client.go

```diff
@@ -653,6 +678,12 @@ func (c *client) OnUpdates(updates []api.Update) {
 					log.Debugf("Global AS number update, recalculate peers")
 					needUpdatePeersV1 = true
 				}
+
+				if cfgKey.Name == "svc_external_ips" {
+					log.Debugf("Global serviceExternalIPs changed.")
+					c.updateExternalIPNets()
```

Comment:
I think we only every call these at the same time, right here.

Might be nicer if we did something like this:

```go
// Get the valid nets.
nets := c.getExternalIPNets()

// Pass the nets to the route generator. 
c.rg.OnBGPConfigurateUpdate(nets)
```

I think it would remove the need for setting the field on the client struct and then accessing the field from within the route generator. WDYT?

Reply:
No, all `OnBGPConfigurateUpdate` does is call `setRouteForSvc` for each service. Therefore, `setRouteForSvc` needs to know about the whitelisted networks. However, `setRouteForSvc` has other entrypoints, for example from `onSvcAdd`, which is why we need to store the nets in a field on the client struct.

Reply:
Right, I see. So we definitely need to store it somewhere so that it's available for future service updates.

I think it would still be nicer to store that internally within the route generator rather than on the client, since the client doesn't every seem to ever access that field. The client and RG run on separate goroutines, so it would also make the locking story a bit cleaner and easier to understand as well if that state was local to the RG.

Reply:
I agree, this is now stored in the rg not the client.